airflow/serialization/serialized_objects.py

```diff
@@ -770,8 +772,12 @@ def _serialize_node(cls, op: BaseOperator | MappedOperator, include_deps: bool)
 
         # Store all template_fields as they are if there are JSON Serializable
         # If not, store them as strings
+        # And raise an exception if the field is not templateable
+        forbidden_fields = set(inspect.signature(BaseOperator.__init__).parameters.keys())
         if op.template_fields:
             for template_field in op.template_fields:
+                if template_field in forbidden_fields:
+                    raise AirflowException(f"Cannot use {template_field} as template field")
```

Comment:
I think the only use case where we can not template fields is when user wants to template BaseOperator parameters which is something that can't be done. So maybe we can be more informative on the exception here?

Reply:
I updated the exception message to `Cannot template BaseOperator fields: {template_field}`, do you have another suggestion?